File: app/src/main/java/com/tohami/egyptinnovate/dagger/module/ViewModelModule.java
package com.tohami.egyptinnovate.dagger.module;

import androidx.lifecycle.ViewModelProvider;

import com.tohami.egyptinnovate.ui.news.details.viewModel.NewsDetailsViewModelFactory;
import com.tohami.egyptinnovate.ui.news.list.viewModel.NewsListViewModelFactory;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class ViewModelModule {
    @Binds
    abstract ViewModelProvider.Factory bindNewsListViewModelFactory(NewsListViewModelFactory factory);

    @Binds
    abstract ViewModelProvider.Factory bindNewsDetailsViewModelFactory(NewsDetailsViewModelFactory factory);
}

File: app/src/main/java/com/tohami/egyptinnovate/dagger/module/ViewModelModule.kt
package com.tohami.egyptinnovate.dagger.module

import androidx.lifecycle.ViewModelProvider

import com.tohami.egyptinnovate.ui.news.details.viewModel.NewsDetailsViewModelFactory
import com.tohami.egyptinnovate.ui.news.list.viewModel.NewsListViewModelFactory

import dagger.Binds
import dagger.Module

@Module
abstract class ViewModelModule {
    @Binds
    internal abstract fun bindNewsListViewModelFactory(factory: NewsListViewModelFactory): ViewModelProvider.Factory

    @Binds
    internal abstract fun bindNewsDetailsViewModelFactory(factory: NewsDetailsViewModelFactory): ViewModelProvider.Factory
}